Libplanet.MySQLStore/MySQLKeyValueStore.cs

```diff
@@ -0,0 +1,216 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Libplanet.Store.Trie;
+using MySqlConnector;
+using MySqlStore.Models;
+using Serilog;
+using SqlKata.Compilers;
+using SqlKata.Execution;
+
+namespace Libplanet.MySqlStore
+{
+    /// <summary>
+    /// The <a href="https://www.mysql.com/">MySql</a> <see cref="IStore"/> implementation.
+    /// This stores data in MySql.
+    /// </summary>
+    public class MySqlKeyValueStore : IKeyValueStore, IDisposable
+    {
+        private readonly QueryFactory? _keyValueDb;
+        private readonly MySqlCompiler _compiler;
+        private readonly string _connectionString;
+        private readonly ILogger _logger;
+        private readonly string _dbName;
+
+        /// <summary>
+        /// Creates a new <see cref="MySqlKeyValueStore"/>.
+        /// </summary>
+        /// <param name="options">The options for creating connection string to MySql.</param>
+        /// <param name="dbName">The name of table in MySql.</param>
+        public MySqlKeyValueStore(MySqlStoreOptions options, string dbName)
+        {
+            var builder = new MySqlConnectionStringBuilder
+            {
+                Database = options.Database,
+                UserID = options.Username,
+                Password = options.Password,
+                Server = options.Server,
+                Port = options.Port,
+            };
+
+            _connectionString = builder.ConnectionString;
+            _compiler = new MySqlCompiler();
+            _logger = Log.ForContext<MySqlKeyValueStore>();
+            _dbName = dbName;
+            _keyValueDb = MySqlUtils.OpenMySqlDB(_connectionString, _compiler);
+        }
+
+        /// <inheritdoc/>
+        public byte[] Get(byte[] key)
+        {
+            var value = Select<KeyValueModel, byte[]>(_dbName, "Key", key);
+
+            if (!value.Any())
+            {
+                throw new KeyNotFoundException(
+                    $"There were no elements that correspond to the key (hex: {ByteUtil.Hex(key)}).");
+            }
+            else
+            {
+                return value[0].Value;
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Set(byte[] key, byte[] value)
+        {
+            Insert(
+                _dbName,
+                new Dictionary<string, object>
+                {
+                    ["Key"] = key,
+                    ["Value"] = value,
+                },
+                "Key",
+                key);
+        }
+
+        /// <inheritdoc/>
+        public void Delete(byte[] key)
+        {
+            Delete(_dbName, "Key", key);
+        }
+
+        public void Dispose()
+        {
+            QueryFactory db = new QueryFactory(new MySqlConnection(_connectionString), _compiler);
+            db.Statement($"truncate table {_dbName}");
```

Comment:
Please refer to https://github.com/planetarium/libplanet-explorer/pull/146#discussion_r555465579

Reply:
If it removes all rows of the `_dbName` table, it will work like [`Libplanet.Tests.MemoryKeyValueStore`](https://github.com/planetarium/libplanet/blob/b10a550/Libplanet.Tests/Store/Trie/MemoryKeyValueStore.cs) which keeps key values while the `MySqlKeyValueStore` keeps alive. For instance, if you use `MySqlKeyValueStore` as argument of `TrieStateStore` then it will store states at `MySqlKeyValueStore`. But, when it was disposed, the all rows will be removed and it will be unable to find the key-value at next time. Is it your intention?